ldes-server-retention/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/retention/services/retentionpolicy/execution/RetentionService.java

```diff
@@ -28,32 +35,27 @@ public RetentionService(MemberPropertiesRepository memberPropertiesRepository, M
 
 	@Scheduled(cron = RETENTION_CRON_KEY)
 	public void executeRetentionPolicies() {
+		log.atDebug().log("Start retention");
 		retentionPolicyCollection
-				.getRetentionPolicyMap()
-				.entrySet()
-				.stream()
-				.filter(this::viewHasRetentionPolicies)
-				.forEach(viewWithRetentionPolicies -> removeMembersFromViewThatMatchRetentionPolicies(
-						viewWithRetentionPolicies.getKey(), viewWithRetentionPolicies.getValue()));
-	}
-
-	private boolean viewHasRetentionPolicies(Map.Entry<String, List<RetentionPolicy>> entry) {
-		return !entry.getValue().isEmpty();
-	}
-
-	private void removeMembersFromViewThatMatchRetentionPolicies(String viewName,
-			List<RetentionPolicy> retentionPoliciesOfView) {
-		memberPropertiesRepository.getMemberPropertiesWithViewReference(viewName)
-				.filter(memberProperties -> memberMatchesAllRetentionPoliciesOfView(retentionPoliciesOfView, viewName,
-						memberProperties))
-				.forEach(memberProperties -> memberRemover.removeMemberFromView(memberProperties, viewName));
+                .getRetentionPolicyMap()
+                .forEach(this::removeMembersFromViewThatMatchRetentionPolicies);
+		log.atDebug().log("Finish retention");
```

Comment:
out of scope of this story.. but this should prefferably be replaced/enriched with tracing.

Reply:
We can add this when the tracing is in place